src/testers.jl

```diff
@@ -100,24 +100,22 @@ function test_frule(
     isapprox_kwargs = (; rtol=rtol, atol=atol, kwargs...)
 
     @testset "test_frule: $f on $(_string_typeof(args))" begin
-        _ensure_not_running_on_functor(f, "test_frule")
-
         xẋs = auto_primal_and_tangent.(args)
         xs = primal.(xẋs)
-        ẋs = tangent.(xẋs)
+        tangents = (rand_tangent(f), tangent.(xẋs)...)
```

Comment:
We should make  `f` also participage in `auto_primal_and_tangents`
and for the functors one would optionally write: `test_frule(f ⊢  ḟ, ...)`

Reply:
Yes, we do.
KeyedArrays use call overloading for accessing by key rather than by integer index.
And for those we likely want to use another KeyedArray as the cotangent

Reply:
Done, and ready for review